### ISSUE TITLE
[JENKINS-49757] Git plugin calls fetch twice per checkout

### ISSUE THREAD

**Issue by jenkins-infra-bot** [jenkinsci/git-plugin#3269]
<p>When performing a git clone operation,</p>

<p>git-client and git-plugin both triggers a git fetch.</p>

<p> </p>

<p>Below are the two lines of code running in the same git clone command both doing fetches.</p>

<p> </p>

<p>The second link is part of the GITSCM step, it triggers the git-client-plugin's CloneCommand (first link) to perform the clone operation. And afterwards, the GITSCM step performs a fetch once the CloneCommand Completes.</p>

<p>However, within the CloneCommand (first link) itself, a fetch command was already done.</p>

<p> </p>

<p><a href="https://github.com/jenkinsci/git-client-plugin/blob/4241e818dbe27fb78093094f29ebcd2203166610/src/main/java/org/jenkinsci/plugins/gitclient/CliGitAPIImpl.java#L609" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/git-client-plugin/blob/4241e818dbe27fb78093094f29ebcd2203166610/src/main/java/org/jenkinsci/plugins/gitclient/CliGitAPIImpl.java#L609</a></p>

<p> </p>

<p><a href="https://github.com/jenkinsci/git-plugin/blob/cb58c74efafafd08bc03bba2787997c6a0272df0/src/main/java/hudson/plugins/git/GitSCM.java#L1153" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/git-plugin/blob/cb58c74efafafd08bc03bba2787997c6a0272df0/src/main/java/hudson/plugins/git/GitSCM.java#L1153</a></p>

<p> </p>

<p>I am aware there are needs to support for git clone to fetch all refs first -<del><a href="https://issues.jenkins.io/browse/JENKINS-31393" title="Git plugin fetches all branches before fetching the specified refspec" class="issue-link" data-issue-key="JENKINS-31393"><del>JENKINS-31393</del></a></del>-. The suggestion provided in that ticket was to use "honor git refspec", so shouldn't "honor git refspec" skip the first fetch here?</p>

<p> </p>

<p>Thanks,</p>

<p>Sam</p>

---
<details><summary><i>Originally reported by <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=slcy">slcy</a>, imported from: <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-49757" target="_blank">Git plugin calls fetch twice per checkout</a></i></summary>
<i><ul>
<li><b>status</b>: Closed
<li><b>priority</b>: Minor
<li><b>component(s)</b>: git-plugin
<li><b>resolution</b>: Fixed
<li><b>resolved</b>: 2020-07-12T01:10:57+00:00
<li><b>votes</b>: 2
<li><b>watchers</b>: 8
<li><b>imported</b>: 2025-12-02
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>When performing a git clone operation,</p>

<p>git-client and git-plugin both triggers a git fetch.</p>

<p> </p>

<p>Below are the two lines of code running in the same git clone command both doing fetches.</p>

<p> </p>

<p>The second link is part of the GITSCM step, it triggers the git-client-plugin's CloneCommand (first link) to perform the clone operation. And afterwards, the GITSCM step performs a fetch once the CloneCommand Completes.</p>

<p>However, within the CloneCommand (first link) itself, a fetch command was already done.</p>

<p> </p>

<p><a href="https://github.com/jenkinsci/git-client-plugin/blob/4241e818dbe27fb78093094f29ebcd2203166610/src/main/java/org/jenkinsci/plugins/gitclient/CliGitAPIImpl.java#L609" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/git-client-plugin/blob/4241e818dbe27fb78093094f29ebcd2203166610/src/main/java/org/jenkinsci/plugins/gitclient/CliGitAPIImpl.java#L609</a></p>

<p> </p>

<p><a href="https://github.com/jenkinsci/git-plugin/blob/cb58c74efafafd08bc03bba2787997c6a0272df0/src/main/java/hudson/plugins/git/GitSCM.java#L1153" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/git-plugin/blob/cb58c74efafafd08bc03bba2787997c6a0272df0/src/main/java/hudson/plugins/git/GitSCM.java#L1153</a></p>

<p> </p>

<p>I am aware there are needs to support for git clone to fetch all refs first -<del><a href="https://issues.jenkins.io/browse/JENKINS-31393" title="Git plugin fetches all branches before fetching the specified refspec" class="issue-link" data-issue-key="JENKINS-31393"><del>JENKINS-31393</del></a></del>-. The suggestion provided in that ticket was to use "honor git refspec", so shouldn't "honor git refspec" skip the first fetch here?</p>

<p> </p>

<p>Thanks,</p>

<p>Sam</p></pre>
</details>
</details>

**Comment by jenkins-infra-bot** [comment]
- _Remote link associated with this issue:_


  - [PR 845 - remove redundant fetch](https://github.com/jenkinsci/git-plugin/pull/845)

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/markewaite.png" title="markewaite's avatar" /> markewaite:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-49757?focusedId=330343&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-330343">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I appreciate the analysis very much.</p>

<p>Unfortunately, the analysis doesn't seem (to me) like evidence of a bug.  The git plugin (your second link) uses the git client plugin (your first link) to perform the fetch.  I don't see from your description that two fetch operations are being performed for each repository being cloned.</p>

<p>Please provide log output which shows two fetches being performed.</p>

<p>Since you've provided those call sites, I assume you could also include stack traces to show that a clone is being performed twice.</p>
</pre>
</details>

</ul>
</details>

<p>I appreciate the analysis very much.</p>

<p>Unfortunately, the analysis doesn't seem (to me) like evidence of a bug.  The git plugin (your second link) uses the git client plugin (your first link) to perform the fetch.  I don't see from your description that two fetch operations are being performed for each repository being cloned.</p>

<p>Please provide log output which shows two fetches being performed.</p>

<p>Since you've provided those call sites, I assume you could also include stack traces to show that a clone is being performed twice.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>slcy:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-49757?focusedId=330346&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-330346">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Hi,</p>

<p> </p>

<p>The clone itself isn't being performed twice, however it's triggering fetch twice. </p>

<p>Both fetches are being trigger by the same Clone operation. Once by the git-plugin after doing a clone, and once more by the git-client-plugin after calling the git-plugin to do the clone.</p>

<p> </p>

<p>Below is the log output</p>

<p> </p>
<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">00:06:32 Cloning repository ssh:<span class="code-comment">//[git project]
</span>00:06:32  &gt; /usr/bin/git init /<span class="code-keyword">var</span>/lib/jenkins/workspace/workspace # timeout=10
00:06:32 Fetching upstream changes from ssh:<span class="code-comment">//[git project]
</span>00:06:32  &gt; /usr/bin/git --version # timeout=10
00:06:32 using GIT_SSH to set credentials Gerrit SSH Key
00:06:32  &gt; /usr/bin/git fetch --tags --progress ssh:<span class="code-comment">//[git project] +refs/heads/*:refs/remotes/origin/* # timeout=20
</span>00:07:49  &gt; /usr/bin/git config remote.origin.url ssh:<span class="code-comment">//[git project] # timeout=10
</span>00:07:49  &gt; /usr/bin/git config --add remote.origin.fetch +refs/heads/*:refs/remotes/origin/* # timeout=10
00:07:49  &gt; /usr/bin/git config remote.origin.url ssh:<span class="code-comment">//[git project] # timeout=10
</span>00:07:49 Fetching upstream changes from ssh:<span class="code-comment">//[git project]
</span>00:07:49 using GIT_SSH to set credentials fpajenkins Gerrit SSH Key
00:07:49  &gt; /usr/bin/git fetch --tags --progress ssh:<span class="code-comment">//[git project] refs/changes/14/3174614/2 # timeout=20
</span>00:08:18  &gt; /usr/bin/git rev-parse cc8266a2cecb79d475d5c3b1dbaa5b2ea3171400^{commit} # timeout=10
00:08:18 Checking out Revision cc8266a2cecb79d475d5c3b1dbaa5b2ea3171400 (detached)
00:08:18  &gt; /usr/bin/git config core.sparsecheckout # timeout=10
00:08:18  &gt; /usr/bin/git config core.sparsecheckout <span class="code-keyword">true</span> # timeout=10
00:08:18  &gt; /usr/bin/git read-tree -mu HEAD # timeout=10
</pre>
</div></div>
<p> </p>
</pre>
</details>

</ul>
</details>

<p>Hi,</p>

<p> </p>

<p>The clone itself isn't being performed twice, however it's triggering fetch twice. </p>

<p>Both fetches are being trigger by the same Clone operation. Once by the git-plugin after doing a clone, and once more by the git-client-plugin after calling the git-plugin to do the clone.</p>

<p> </p>

<p>Below is the log output</p>

<p> </p>

<pre>
00:06:32 Cloning repository ssh:<span class="code-comment">//[git project]
</span>00:06:32  > /usr/bin/git init /<span class="code-keyword">var</span>/lib/jenkins/workspace/workspace # timeout=10
00:06:32 Fetching upstream changes from ssh:<span class="code-comment">//[git project]
</span>00:06:32  > /usr/bin/git --version # timeout=10
00:06:32 using GIT_SSH to set credentials Gerrit SSH Key
00:06:32  > /usr/bin/git fetch --tags --progress ssh:<span class="code-comment">//[git project] +refs/heads/*:refs/remotes/origin/* # timeout=20
</span>00:07:49  > /usr/bin/git config remote.origin.url ssh:<span class="code-comment">//[git project] # timeout=10
</span>00:07:49  > /usr/bin/git config --add remote.origin.fetch +refs/heads/*:refs/remotes/origin/* # timeout=10
00:07:49  > /usr/bin/git config remote.origin.url ssh:<span class="code-comment">//[git project] # timeout=10
</span>00:07:49 Fetching upstream changes from ssh:<span class="code-comment">//[git project]
</span>00:07:49 using GIT_SSH to set credentials fpajenkins Gerrit SSH Key
00:07:49  > /usr/bin/git fetch --tags --progress ssh:<span class="code-comment">//[git project] refs/changes/14/3174614/2 # timeout=20
</span>00:08:18  > /usr/bin/git rev-parse cc8266a2cecb79d475d5c3b1dbaa5b2ea3171400^{commit} # timeout=10
00:08:18 Checking out Revision cc8266a2cecb79d475d5c3b1dbaa5b2ea3171400 (detached)
00:08:18  > /usr/bin/git config core.sparsecheckout # timeout=10
00:08:18  > /usr/bin/git config core.sparsecheckout <span class="code-keyword">true</span> # timeout=10
00:08:18  > /usr/bin/git read-tree -mu HEAD # timeout=10
</pre>
<p> </p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>slcy:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-49757?focusedId=330347&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-330347">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>If you look at the first link. It's calling the fetchcommand.execute directly and skipping all the fetchCommand decorators.</p>
</pre>
</details>

</ul>
</details>

<p>If you look at the first link. It's calling the fetchcommand.execute directly and skipping all the fetchCommand decorators.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>slcy:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-49757?focusedId=330348&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-330348">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=markewaite" class="user-hover" rel="markewaite">markewaite</a></p>

<p>The second link is part of the GITSCM step, it triggers the git-client-plugin's CloneCommand (first link) to perform the clone operation. And afterwards, the GITSCM step performs a fetch once the CloneCommand Completes.</p>

<p>However, within the CloneCommand (first link) itself, a fetch command was already done.</p>

<p> </p>

<p>see GitSCM#1140 on the call from GITSCM into CloneCommand,</p>

<p>see CliGitAPIImpl#609 inside CloneCommand for the first fetch.</p>

<p>see GitSCM#1153 for the second fetch</p>

<p> </p>

<p>Thanks a lot for looking into this</p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=markewaite" class="user-hover" rel="markewaite">markewaite</a></p>

<p>The second link is part of the GITSCM step, it triggers the git-client-plugin's CloneCommand (first link) to perform the clone operation. And afterwards, the GITSCM step performs a fetch once the CloneCommand Completes.</p>

<p>However, within the CloneCommand (first link) itself, a fetch command was already done.</p>

<p> </p>

<p>see GitSCM#1140 on the call from GITSCM into CloneCommand,</p>

<p>see CliGitAPIImpl#609 inside CloneCommand for the first fetch.</p>

<p>see GitSCM#1153 for the second fetch</p>

<p> </p>

<p>Thanks a lot for looking into this</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/markewaite.png" title="markewaite's avatar" /> markewaite:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-49757?focusedId=330349&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-330349">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I think I see the same behavior in the log file of my jobs.  I have a <a href="https://github.com/MarkEWaite/docker-lfs/blob/lts-with-plugins/ref/jobs/Bugs-Individual/jobs/JENKINS-48064-Fisheye-url-validator-ignored/config.xml" class="external-link" target="_blank" rel="nofollow noopener">Freestyle job</a> configured which generates the following log output:</p>

<div class="preformatted panel" style="border-width: 1px;"><div class="preformattedContent panelContent">
<pre>&gt; git init /home/beemarkwaite/mark-pc2.markwaite.net-agent/workspace/Bugs-Individual/JENKINS-48064-Fisheye-url-validator-ignored # timeout=10
Fetching upstream changes from https://github.com/aeshell/aesh.git
 &gt; git --version # timeout=10
 &gt; git fetch --tags --progress https://github.com/aeshell/aesh.git +refs/heads/master:refs/remotes/origin/master +refs/heads/0.56.1:refs/remotes/origin/0.56.1
 &gt; git config remote.origin.url https://github.com/aeshell/aesh.git # timeout=10
 &gt; git config --add remote.origin.fetch +refs/heads/master:refs/remotes/origin/master # timeout=10
 &gt; git config --add remote.origin.fetch +refs/heads/0.56.1:refs/remotes/origin/0.56.1 # timeout=10
 &gt; git config remote.origin.url https://github.com/aeshell/aesh.git # timeout=10
Fetching upstream changes from https://github.com/aeshell/aesh.git
 &gt; git fetch --tags --progress https://github.com/aeshell/aesh.git +refs/heads/master:refs/remotes/origin/master +refs/heads/0.56.1:refs/remotes/origin/0.56.1
</pre>
</div></div>

<p>That shows two calls are made to <tt>git fetch</tt>.  Since <tt>git fetch</tt> is an incremental operation, I'm not planning to make any change in this area of the code.  The second fetch operation is not free, but it is usually very lightweight, since it will only request changes made since the preceding fetch.</p>

<p>For the sake of precision, your earlier comments are not describing the implementation of the code.  You said:</p>

<blockquote><p>The first link is part of the Git-client clone step, it triggers the git-plugin CloneCommand (second link) to perform the clone operations. And afterwards, the Git-client plugin performs a fetch.</p></blockquote>

<p>As far as I can tell, that's not the way the code works.  The git client plugin does not call the git plugin.  It does not have any dependency on the git plugin and does not know about the git plugin implementation.  The git client plugin provides an API that is called by the git plugin.</p>

<p>The stack trace you provided shows two calls to <tt>git fetch</tt>, but the two calls are using different credentials.  I'm assuming that the difference in credentials is because you edited the log output to remove  identifying information.</p>
</pre>
</details>

</ul>
</details>

<p>I think I see the same behavior in the log file of my jobs.  I have a <a href="https://github.com/MarkEWaite/docker-lfs/blob/lts-with-plugins/ref/jobs/Bugs-Individual/jobs/JENKINS-48064-Fisheye-url-validator-ignored/config.xml" class="external-link" target="_blank" rel="nofollow noopener">Freestyle job</a> configured which generates the following log output:</p>



```
> git init /home/beemarkwaite/mark-pc2.markwaite.net-agent/workspace/Bugs-Individual/JENKINS-48064-Fisheye-url-validator-ignored # timeout=10
Fetching upstream changes from https://github.com/aeshell/aesh.git
 > git --version # timeout=10
 > git fetch --tags --progress https://github.com/aeshell/aesh.git +refs/heads/master:refs/remotes/origin/master +refs/heads/0.56.1:refs/remotes/origin/0.56.1
 > git config remote.origin.url https://github.com/aeshell/aesh.git # timeout=10
 > git config --add remote.origin.fetch +refs/heads/master:refs/remotes/origin/master # timeout=10
 > git config --add remote.origin.fetch +refs/heads/0.56.1:refs/remotes/origin/0.56.1 # timeout=10
 > git config remote.origin.url https://github.com/aeshell/aesh.git # timeout=10
Fetching upstream changes from https://github.com/aeshell/aesh.git
 > git fetch --tags --progress https://github.com/aeshell/aesh.git +refs/heads/master:refs/remotes/origin/master +refs/heads/0.56.1:refs/remotes/origin/0.56.1

```

<p>That shows two calls are made to <tt>git fetch</tt>.  Since <tt>git fetch</tt> is an incremental operation, I'm not planning to make any change in this area of the code.  The second fetch operation is not free, but it is usually very lightweight, since it will only request changes made since the preceding fetch.</p>

<p>For the sake of precision, your earlier comments are not describing the implementation of the code.  You said:</p>

<blockquote><p>The first link is part of the Git-client clone step, it triggers the git-plugin CloneCommand (second link) to perform the clone operations. And afterwards, the Git-client plugin performs a fetch.</p></blockquote>

<p>As far as I can tell, that's not the way the code works.  The git client plugin does not call the git plugin.  It does not have any dependency on the git plugin and does not know about the git plugin implementation.  The git client plugin provides an API that is called by the git plugin.</p>

<p>The stack trace you provided shows two calls to <tt>git fetch</tt>, but the two calls are using different credentials.  I'm assuming that the difference in credentials is because you edited the log output to remove  identifying information.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/dogin2006.svg" title="dogin2006's avatar" /> dogin2006:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-49757?focusedId=337423&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-337423">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Facing same issue when using gerrit trigger.</p>

<p>My solution is using pipeline job like below</p>
<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">stage(<span class="code-quote">'CheckOut'</span>) {
    steps {
        sh <span class="code-quote">"git init"</span>
        checkout(scm)
    }
}

post {
    cleanup {
        deleteDir()
    }
}</pre>
</div></div>
<p>Init git repo first will skip clone step.</p>

<p>Do not use WipeWorkspace in scm step, clean workspace in post closure</p>
</pre>
</details>

</ul>
</details>

<p>Facing same issue when using gerrit trigger.</p>

<p>My solution is using pipeline job like below</p>

<pre>
stage(<span class="code-quote">'CheckOut'</span>) {
    steps {
sh <span class="code-quote">"git init"</span>
checkout(scm)
    }
}

post {
    cleanup {
deleteDir()
    }
}</pre>
<p>Init git repo first will skip clone step.</p>

<p>Do not use WipeWorkspace in scm step, clean workspace in post closure</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/markewaite.png" title="markewaite's avatar" /> markewaite:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-49757?focusedId=337424&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-337424">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=dogin2006" class="user-hover" rel="dogin2006">dogin2006</a> I believe you are seeing a different condition.  The `sh 'git iniit'` initializes an empty git repository in the workspace.  The `checkout(scm)` step likely erases the newly created empty git repository and creates a new git repository populated with the results of fetching changes from the remote server.</p>

<p>The  deleteDir() step removes the workspace contents at the end.</p>

<p>As far as I understand declarative Pipeline, it will perform the initial checkout implicitly (unless you'v added the skipDefaultCheckout property), then it sees your `checkout scm` and does it again.</p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=dogin2006" class="user-hover" rel="dogin2006">dogin2006</a> I believe you are seeing a different condition.  The `sh 'git iniit'` initializes an empty git repository in the workspace.  The `checkout(scm)` step likely erases the newly created empty git repository and creates a new git repository populated with the results of fetching changes from the remote server.</p>

<p>The  deleteDir() step removes the workspace contents at the end.</p>

<p>As far as I understand declarative Pipeline, it will perform the initial checkout implicitly (unless you'v added the skipDefaultCheckout property), then it sees your `checkout scm` and does it again.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/dogin2006.svg" title="dogin2006's avatar" /> dogin2006:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-49757?focusedId=337431&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-337431">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=markewaite" class="user-hover" rel="markewaite">markewaite</a></p>

<p>First of all, my job is using Gerrit Trigger.</p>

<p>When my Jenkinsfile is like</p>

<p> </p>
<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">def scm = [
        $class                           : <span class="code-quote">'GitSCM'</span>,
        branches                         : [[name: <span class="code-quote">'*/master'</span>]],
        doGenerateSubmoduleConfigurations: <span class="code-keyword">false</span>,
        extensions                       : [
                [$class: <span class="code-quote">'WipeWorkspace'</span>],
                [$class: <span class="code-quote">'CloneOption'</span>, depth: 2, honorRefspec: <span class="code-keyword">true</span>, noTags: <span class="code-keyword">true</span>, reference: '', shallow: <span class="code-keyword">true</span>, timeout: 10],
                [$class: <span class="code-quote">'BuildChooserSetting'</span>, buildChooser: [$class: <span class="code-quote">'GerritTriggerBuildChooser'</span>]]
        ],
        submoduleCfg                     : [],
        userRemoteConfigs                : [
                [
                        credentialsId: <span class="code-quote">'hide'</span>,
                        refspec      : <span class="code-quote">'$GERRIT_REFSPEC'</span>,
                        url          : <span class="code-quote">'hide'</span>
                ]
        ]
]

pipeline {
    agent {
        node {
            label <span class="code-quote">'gerrit'</span>
        }
    }
    options {
        skipDefaultCheckout <span class="code-keyword">true</span>
    }
    stages {
        stage(<span class="code-quote">'CheckOut'</span>) {
            steps {
                checkout(scm)
            }
        }
    }
}
</pre>
</div></div>
<p>The console log</p>

<p> </p>

<p> </p>
<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">Wiping out workspace first.
Cloning the remote Git repository
Using shallow clone
shallow clone depth 2
Avoid fetching tags
Honoring refspec on initial clone
Cloning repository hide
&gt; git init hide # timeout=10
Fetching upstream changes from hide
&gt; git --version # timeout=10
using GIT_SSH to set credentials
&gt; git fetch --no-tags --progress hide $GERRIT_REFSPEC --depth=2 # timeout=10
&gt; git config remote.origin.url hide # timeout=10
&gt; git config --add remote.origin.fetch $GERRIT_REFSPEC # timeout=10
&gt; git config remote.origin.url hide # timeout=10
Fetching upstream changes from hide
using GIT_SSH to set credentials
&gt; git fetch --no-tags --progress hide refs/changes/64/4264/7 --depth=2 # timeout=10
&gt; git rev-parse 4b5226960edd4551bbafc058db4afc0e69bd103f^{commit} # timeout=10
Checking out Revision 4b5226960edd4551bbafc058db4afc0e69bd103f (refs/changes/64/4264/7)
&gt; git config core.sparsecheckout # timeout=10
&gt; git checkout -f 4b5226960edd4551bbafc058db4afc0e69bd103f
Commit message: <span class="code-quote">"hide"</span>
&gt; git rev-parse FETCH_HEAD^{commit} # timeout=10
&gt; git rev-list --no-walk a1323feb6fa57ce13c87a5fd6886b14aa3f75dab # timeout=10
</pre>
</div></div>
<p>Fetch twice and the first fetch cannot recognize $GERRIT_REFSPEC</p>

<p>Then I changed my Jenkinsfile like</p>

<p> </p>
<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">def scm = [
        $class                           : <span class="code-quote">'GitSCM'</span>,
        branches                         : [[name: <span class="code-quote">'*/master'</span>]],
        doGenerateSubmoduleConfigurations: <span class="code-keyword">false</span>,
        extensions                       : [
                [$class: <span class="code-quote">'CloneOption'</span>, depth: 2, honorRefspec: <span class="code-keyword">true</span>, noTags: <span class="code-keyword">true</span>, reference: '', shallow: <span class="code-keyword">true</span>, timeout: 10],
                [$class: <span class="code-quote">'BuildChooserSetting'</span>, buildChooser: [$class: <span class="code-quote">'GerritTriggerBuildChooser'</span>]]
        ],
        submoduleCfg                     : [],
        userRemoteConfigs                : [
                [
                        credentialsId: <span class="code-quote">'hide'</span>,
                        refspec      : <span class="code-quote">'$GERRIT_REFSPEC'</span>,
                        url          : <span class="code-quote">'hide'</span>
                ]
        ]
]

pipeline {
    agent {
        node {
            label <span class="code-quote">'gerrit'</span>
        }
    }
    options {
        skipDefaultCheckout <span class="code-keyword">true</span>
    }
    stages {
        stage(<span class="code-quote">'CheckOut'</span>) {
            steps {
                sh(<span class="code-quote">'git init'</span>)
                checkout(scm)
            }
        }
    }
    post {
        cleanup {
            cleanWs()
        }
    }
}
</pre>
</div></div>
<p>The console log</p>

<p> </p>

<p> </p>
<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">[gerrit_test_2] Running shell script
+ git init
Initialized empty Git repository in hide
[Pipeline] checkout
 &gt; git rev-parse --is-inside-work-tree # timeout=10
Fetching changes from the remote Git repository
 &gt; git config remote.origin.url hide # timeout=10
Fetching upstream changes from hide
 &gt; git --version # timeout=10
using GIT_SSH to set credentials 
 &gt; git fetch --no-tags --progress hide refs/changes/64/4264/7 --depth=2 # timeout=10
 &gt; git rev-parse 4b5226960edd4551bbafc058db4afc0e69bd103f^{commit} # timeout=10
Checking out Revision 4b5226960edd4551bbafc058db4afc0e69bd103f (refs/changes/64/4264/7)
 &gt; git config core.sparsecheckout # timeout=10
 &gt; git checkout -f 4b5226960edd4551bbafc058db4afc0e69bd103f
Commit message: <span class="code-quote">"hide"</span>
 &gt; git rev-parse FETCH_HEAD^{commit} # timeout=10
 &gt; git rev-list --no-walk a1323feb6fa57ce13c87a5fd6886b14aa3f75dab # timeout=10
[Pipeline] }
[Pipeline] <span class="code-comment">// stage
</span>[Pipeline] stage
[Pipeline] { (Declarative: Post Actions)
[Pipeline] cleanWs
[WS-CLEANUP] Deleting project workspace...[WS-CLEANUP] done
[Pipeline] }
[Pipeline] <span class="code-comment">// stage
</span>[Pipeline] }
[Pipeline] <span class="code-comment">// node
</span>[Pipeline] End of Pipeline
Finished: SUCCESS
</pre>
</div></div>
<p>Fetch only once</p>

<p> </p>

<p> </p>

<p>Jenkins Version: 2.120</p>

<p>Gerrit Trigger: 2.27.5</p>

<p>Git client plugin: 2.7.1</p>

<p>Git plugin: 3.8.0</p>

<p> </p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=markewaite" class="user-hover" rel="markewaite">markewaite</a></p>

<p>First of all, my job is using Gerrit Trigger.</p>

<p>When my Jenkinsfile is like</p>

<p> </p>

<pre>
def scm = [
$class   : <span class="code-quote">'GitSCM'</span>,
branches : [[name: <span class="code-quote">'*/master'</span>]],
doGenerateSubmoduleConfigurations: <span class="code-keyword">false</span>,
extensions       : [
[$class: <span class="code-quote">'WipeWorkspace'</span>],
[$class: <span class="code-quote">'CloneOption'</span>, depth: 2, honorRefspec: <span class="code-keyword">true</span>, noTags: <span class="code-keyword">true</span>, reference: '', shallow: <span class="code-keyword">true</span>, timeout: 10],
[$class: <span class="code-quote">'BuildChooserSetting'</span>, buildChooser: [$class: <span class="code-quote">'GerritTriggerBuildChooser'</span>]]
],
submoduleCfg     : [],
userRemoteConfigs: [
[
credentialsId: <span class="code-quote">'hide'</span>,
refspec      : <span class="code-quote">'$GERRIT_REFSPEC'</span>,
url  : <span class="code-quote">'hide'</span>
]
]
]

pipeline {
    agent {
node {
    label <span class="code-quote">'gerrit'</span>
}
    }
    options {
skipDefaultCheckout <span class="code-keyword">true</span>
    }
    stages {
stage(<span class="code-quote">'CheckOut'</span>) {
    steps {
checkout(scm)
    }
}
    }
}
</pre>
<p>The console log</p>

<p> </p>

<p> </p>

<pre>
Wiping out workspace first.
Cloning the remote Git repository
Using shallow clone
shallow clone depth 2
Avoid fetching tags
Honoring refspec on initial clone
Cloning repository hide
> git init hide # timeout=10
Fetching upstream changes from hide
> git --version # timeout=10
using GIT_SSH to set credentials
> git fetch --no-tags --progress hide $GERRIT_REFSPEC --depth=2 # timeout=10
> git config remote.origin.url hide # timeout=10
> git config --add remote.origin.fetch $GERRIT_REFSPEC # timeout=10
> git config remote.origin.url hide # timeout=10
Fetching upstream changes from hide
using GIT_SSH to set credentials
> git fetch --no-tags --progress hide refs/changes/64/4264/7 --depth=2 # timeout=10
> git rev-parse 4b5226960edd4551bbafc058db4afc0e69bd103f^{commit} # timeout=10
Checking out Revision 4b5226960edd4551bbafc058db4afc0e69bd103f (refs/changes/64/4264/7)
> git config core.sparsecheckout # timeout=10
> git checkout -f 4b5226960edd4551bbafc058db4afc0e69bd103f
Commit message: <span class="code-quote">"hide"</span>
> git rev-parse FETCH_HEAD^{commit} # timeout=10
> git rev-list --no-walk a1323feb6fa57ce13c87a5fd6886b14aa3f75dab # timeout=10
</pre>
<p>Fetch twice and the first fetch cannot recognize $GERRIT_REFSPEC</p>

<p>Then I changed my Jenkinsfile like</p>

<p> </p>

<pre>
def scm = [
$class   : <span class="code-quote">'GitSCM'</span>,
branches : [[name: <span class="code-quote">'*/master'</span>]],
doGenerateSubmoduleConfigurations: <span class="code-keyword">false</span>,
extensions       : [
[$class: <span class="code-quote">'CloneOption'</span>, depth: 2, honorRefspec: <span class="code-keyword">true</span>, noTags: <span class="code-keyword">true</span>, reference: '', shallow: <span class="code-keyword">true</span>, timeout: 10],
[$class: <span class="code-quote">'BuildChooserSetting'</span>, buildChooser: [$class: <span class="code-quote">'GerritTriggerBuildChooser'</span>]]
],
submoduleCfg     : [],
userRemoteConfigs: [
[
credentialsId: <span class="code-quote">'hide'</span>,
refspec      : <span class="code-quote">'$GERRIT_REFSPEC'</span>,
url  : <span class="code-quote">'hide'</span>
]
]
]

pipeline {
    agent {
node {
    label <span class="code-quote">'gerrit'</span>
}
    }
    options {
skipDefaultCheckout <span class="code-keyword">true</span>
    }
    stages {
stage(<span class="code-quote">'CheckOut'</span>) {
    steps {
sh(<span class="code-quote">'git init'</span>)
checkout(scm)
    }
}
    }
    post {
cleanup {
    cleanWs()
}
    }
}
</pre>
<p>The console log</p>

<p> </p>

<p> </p>

<pre>
[gerrit_test_2] Running shell script
+ git init
Initialized empty Git repository in hide
[Pipeline] checkout
 > git rev-parse --is-inside-work-tree # timeout=10
Fetching changes from the remote Git repository
 > git config remote.origin.url hide # timeout=10
Fetching upstream changes from hide
 > git --version # timeout=10
using GIT_SSH to set credentials 
 > git fetch --no-tags --progress hide refs/changes/64/4264/7 --depth=2 # timeout=10
 > git rev-parse 4b5226960edd4551bbafc058db4afc0e69bd103f^{commit} # timeout=10
Checking out Revision 4b5226960edd4551bbafc058db4afc0e69bd103f (refs/changes/64/4264/7)
 > git config core.sparsecheckout # timeout=10
 > git checkout -f 4b5226960edd4551bbafc058db4afc0e69bd103f
Commit message: <span class="code-quote">"hide"</span>
 > git rev-parse FETCH_HEAD^{commit} # timeout=10
 > git rev-list --no-walk a1323feb6fa57ce13c87a5fd6886b14aa3f75dab # timeout=10
[Pipeline] }
[Pipeline] <span class="code-comment">// stage
</span>[Pipeline] stage
[Pipeline] { (Declarative: Post Actions)
[Pipeline] cleanWs
[WS-CLEANUP] Deleting project workspace...[WS-CLEANUP] done
[Pipeline] }
[Pipeline] <span class="code-comment">// stage
</span>[Pipeline] }
[Pipeline] <span class="code-comment">// node
</span>[Pipeline] End of Pipeline
Finished: SUCCESS
</pre>
<p>Fetch only once</p>

<p> </p>

<p> </p>

<p>Jenkins Version: 2.120</p>

<p>Gerrit Trigger: 2.27.5</p>

<p>Git client plugin: 2.7.1</p>

<p>Git plugin: 3.8.0</p>

<p> </p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/markewaite.png" title="markewaite's avatar" /> markewaite:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-49757?focusedId=337450&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-337450">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=dogin2006" class="user-hover" rel="dogin2006">dogin2006</a> since you're using Pipeline, you can expand the value of GERRIT_REFSPEC without relying on the git plugin internal environment variable expansion.  Use: </p>

<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">refspec      : <span class="code-quote">"${GERRIT_REFSPEC}"</span>,
</pre>
</div></div>

<p>instead of </p>

<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">refspec      : <span class="code-quote">'$GERRIT_REFSPEC'</span>,
</pre>
</div></div>

<p>The bug report is specific to Freestyle jobs because the variable expansion for strings is not available to Freestyle jobs.</p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=dogin2006" class="user-hover" rel="dogin2006">dogin2006</a> since you're using Pipeline, you can expand the value of GERRIT_REFSPEC without relying on the git plugin internal environment variable expansion.  Use: </p>


<pre>
refspec      : <span class="code-quote">"${GERRIT_REFSPEC}"</span>,
</pre>

<p>instead of </p>


<pre>
refspec      : <span class="code-quote">'$GERRIT_REFSPEC'</span>,
</pre>

<p>The bug report is specific to Freestyle jobs because the variable expansion for strings is not available to Freestyle jobs.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/dogin2006.svg" title="dogin2006's avatar" /> dogin2006:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-49757?focusedId=337510&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-337510">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=markewaite" class="user-hover" rel="markewaite">markewaite</a></p>

<p>Using "${GERRIT_REFSPEC}" fix the first problem.</p>

<p> </p>
<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">10:38:45 Cloning the remote Git repository
10:38:45 Using shallow clone
10:38:45 shallow clone depth 2
10:38:45 Avoid fetching tags
10:38:45 Honoring refspec on initial clone
10:38:45 Cloning repository hide
10:38:45 &gt; git init hide # timeout=10
10:38:45 Fetching upstream changes from hide
10:38:45 &gt; git --version # timeout=10
10:38:45 using GIT_SSH to set credentials 
10:38:45 &gt; git fetch --no-tags --progress hide refs/changes/50/4750/1 --depth=2 # timeout=10
10:39:34 &gt; git config remote.origin.url hide # timeout=10
10:39:34 &gt; git config --add remote.origin.fetch refs/changes/50/4750/1 # timeout=10
10:39:34 &gt; git config remote.origin.url hide # timeout=10
10:39:34 Fetching upstream changes from hide
10:39:34 using GIT_SSH to set credentials 
10:39:34 &gt; git fetch --no-tags --progress hide refs/changes/50/4750/1 --depth=2 # timeout=10
10:40:34 &gt; git rev-parse 44fd5ca794ef1d4ccbfb437d94ef804626dea726^{commit} # timeout=10
10:40:34 Checking out Revision 44fd5ca794ef1d4ccbfb437d94ef804626dea726 (refs/changes/50/4750/1)
10:40:34 &gt; git config core.sparsecheckout # timeout=10
10:40:34 &gt; git checkout -f 44fd5ca794ef1d4ccbfb437d94ef804626dea726 # timeout=10
10:40:38 Commit message: <span class="code-quote">"hide"</span>
10:40:38 &gt; git rev-parse FETCH_HEAD^{commit} # timeout=10
10:40:38 &gt; git rev-list --no-walk eb696053ba8f4c94c388db0334b886879e6aaa42 # timeout=10
</pre>
</div></div>
<p>But it's also triggering fetch twice and the second fetch is not cost free.</p>

<p>In our case, both fetch cost about one minute.</p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=markewaite" class="user-hover" rel="markewaite">markewaite</a></p>

<p>Using "${GERRIT_REFSPEC}" fix the first problem.</p>

<p> </p>

<pre>
10:38:45 Cloning the remote Git repository
10:38:45 Using shallow clone
10:38:45 shallow clone depth 2
10:38:45 Avoid fetching tags
10:38:45 Honoring refspec on initial clone
10:38:45 Cloning repository hide
10:38:45 > git init hide # timeout=10
10:38:45 Fetching upstream changes from hide
10:38:45 > git --version # timeout=10
10:38:45 using GIT_SSH to set credentials 
10:38:45 > git fetch --no-tags --progress hide refs/changes/50/4750/1 --depth=2 # timeout=10
10:39:34 > git config remote.origin.url hide # timeout=10
10:39:34 > git config --add remote.origin.fetch refs/changes/50/4750/1 # timeout=10
10:39:34 > git config remote.origin.url hide # timeout=10
10:39:34 Fetching upstream changes from hide
10:39:34 using GIT_SSH to set credentials 
10:39:34 > git fetch --no-tags --progress hide refs/changes/50/4750/1 --depth=2 # timeout=10
10:40:34 > git rev-parse 44fd5ca794ef1d4ccbfb437d94ef804626dea726^{commit} # timeout=10
10:40:34 Checking out Revision 44fd5ca794ef1d4ccbfb437d94ef804626dea726 (refs/changes/50/4750/1)
10:40:34 > git config core.sparsecheckout # timeout=10
10:40:34 > git checkout -f 44fd5ca794ef1d4ccbfb437d94ef804626dea726 # timeout=10
10:40:38 Commit message: <span class="code-quote">"hide"</span>
10:40:38 > git rev-parse FETCH_HEAD^{commit} # timeout=10
10:40:38 > git rev-list --no-walk eb696053ba8f4c94c388db0334b886879e6aaa42 # timeout=10
</pre>
<p>But it's also triggering fetch twice and the second fetch is not cost free.</p>

<p>In our case, both fetch cost about one minute.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/markewaite.png" title="markewaite's avatar" /> markewaite:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-49757?focusedId=337511&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-337511">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=dogin2006" class="user-hover" rel="dogin2006">dogin2006</a> you're welcome to propose a pull request to the plugin to improve its behavior.  Alternately, you could try replacing the "depth=2" with a reference repository to see if that would reduce the fetch time.</p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=dogin2006" class="user-hover" rel="dogin2006">dogin2006</a> you're welcome to propose a pull request to the plugin to improve its behavior.  Alternately, you could try replacing the "depth=2" with a reference repository to see if that would reduce the fetch time.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/oliverp.svg" title="oliverp's avatar" /> oliverp:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-49757?focusedId=350756&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-350756">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Using a reference repository doesn't fix the issue.</p>

<p>Using the command line git clone including reference repository, it takes approx 11 seconds and the same configuration via the checkout scm option takes 1 minute 3x seconds.</p>

<p> </p>
</pre>
</details>

</ul>
</details>

<p>Using a reference repository doesn't fix the issue.</p>

<p>Using the command line git clone including reference repository, it takes approx 11 seconds and the same configuration via the checkout scm option takes 1 minute 3x seconds.</p>

<p> </p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/rishabhbudhouliya.png" title="rishabhbudhouliya's avatar" /> rishabhbudhouliya:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-49757?focusedId=386275&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-386275">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I have attempted to fix this issue: 
<a href="https://github.com/jenkinsci/git-plugin/pull/845" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/git-plugin/pull/845</a></p>
</pre>
</details>

</ul>
</details>

<p>I have attempted to fix this issue: <br/>
<a href="https://github.com/jenkinsci/git-plugin/pull/845" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/git-plugin/pull/845</a></p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/markewaite.png" title="markewaite's avatar" /> markewaite:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-49757?focusedId=394105&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-394105">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>The fix from <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=rishabhbudhouliya" class="user-hover" rel="rishabhbudhouliya">rishabhbudhouliya</a> has been merged to the master branch and will be included in the next release of the git plugin</p>
</pre>
</details>

</ul>
</details>

<p>The fix from <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=rishabhbudhouliya" class="user-hover" rel="rishabhbudhouliya">rishabhbudhouliya</a> has been merged to the master branch and will be included in the next release of the git plugin</p>

**Comment by jenkins-infra-bot** [comment]
<i>[Original `relates_to` from Jira: <a href="https://github.com/jenkinsci/git-plugin/issues?q=is%3Aissue%20%22jira_issue_key%3DJENKINS-56404%22">JENKINS-56404</a>]</i>
